code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/azure/AzureRESTClient.java

```diff
@@ -237,16 +237,22 @@ public void refreshAccessToken() {
 
     public AzureAccountInfo jsonToAzureAccountInfo(Map<String, Object> jsonData) {
         String objectId = ObjectUtils.toString(jsonData.get("objectId"));
+        String objectType = ObjectUtils.toString(jsonData.get("objectType"));
         String userPrincipalName = ObjectUtils.toString(jsonData.get("userPrincipalName"));
         String accountName = ObjectUtils.toString(jsonData.get("mailNickname"));
+
+        if(objectType != null && objectType.equalsIgnoreCase("Group")) {
```

Comment:
Conditions like these can be simplified by flipping the equals, so
```java
"Group".equalsIgnoreCase(objectType)
```
Then you avoid the NPE or need for a null check.